pkgs/by-name/wa/wally/package.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  openssl,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "wally";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "UpliftGames";
+    repo = "wally";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-lbEUj6iWwm3KtfUwNkJf8cSjXMQ4Mki/jAqQavDajUA=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  # https://github.com/UpliftGames/wally/issues/223
+  cargoPatches = [ ./Cargo.lock.patch ];
+  cargoHash = "sha256-5Lfs5GlOPxvltFpbgPBhStWxPGqv8nYpY/WC2ABtea0=";
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = "--version";
```

Comment:
```suggestion

```
this line isn't required and imo you don't have to remove it but some reviewers will ask you to